tests/test_clientsecrets.py

```diff
@@ -210,32 +210,33 @@ def test_validate_error(self):
             # Ensure that it is unicode
             src = _from_bytes(src)
             # Test load(s)
-            try:
+            with self.assertRaises(
+                    clientsecrets.InvalidClientSecretsError) as exc_manager:
                 clientsecrets.loads(src)
-                self.fail(src + ' should not be a valid client_secrets file.')
-            except clientsecrets.InvalidClientSecretsError as e:
-                self.assertTrue(str(e).startswith(match))
+
+            first_caught_exception = exc_manager.exception
```

Comment:
No need for a local field; use `exc_manager.exception` everywhere you wish now that you have it available for use.